src/metadata/metadata-storage.ts

```diff
@@ -108,11 +113,20 @@ export class MetadataStorage {
     this.fieldDirectives.push(definition);
   }
 
+  collectExtensionsClassMetadata(definition: ExtensionsClassMetadata) {
+    this.classExtensions.push(definition);
+  }
+  collectExtensionsFieldMetadata(definition: ExtensionsFieldMetadata) {
+    this.fieldExtensions.push(definition);
+  }
+
   build() {
     // TODO: disable next build attempts
 
     this.classDirectives.reverse();
     this.fieldDirectives.reverse();
+    this.classExtensions.reverse();
```

Comment:
This is necessary or else, the first decorator declared will have priority, e.g.:

```
@ObjectType
@Extensions({ duplicate: "one" })
@Extensions({ duplicate: "two" })
```

would end up with the object having `duplicate: "one"`, it seems counter intuitive to me and I think this is why it was done in the "directives" feature in the same way. I would have added a code comment but since there was none in the directives PR, I'm not sure it would be appreciated. Will ask the question to the maintainer